src/storage-controller/src/persist_handles.rs

```diff
@@ -70,7 +70,7 @@ pub(crate) enum SnapshotStatsAsOf<T: Timestamp + Lattice + Codec64> {
 enum PersistReadWorkerCmd<T: Timestamp + Lattice + Codec64> {
     Register(GlobalId, SinceHandle<SourceData, (), T, Diff, PersistEpoch>),
     Update(GlobalId, SinceHandle<SourceData, (), T, Diff, PersistEpoch>),
-    Downgrade(BTreeMap<GlobalId, Antichain<T>>),
+    Downgrade(BTreeMap<GlobalId, (Antichain<T>, oneshot::Sender<bool>)>),
```

Comment:
Using a bool here to indicate success or failure feels brittle. I'd at the very least use a `Result<(), ()>` though seems like we could have a meaningful error to pass along

Reply:
Agreed, refactored to use `Result<(), ()>`. We could do a `Result<(), anyhow::Error>` but since downgrade returns an `Option<Result<...>>` and `oneshot::Receiver` can also error, it's a bit easier to "flatten" to `()`